### PR TITLE
fix: apply default client config to all commands

### LINKED ISSUE #38157: Camunda Client doesn't use default tenant id for correlating messages
### 

### 

### 

### 

### Description

When correlating a message via `client().newCorrelateMessageCommand()`, the default client id set on the client `CamundaClient.newClientBuilder().defaultTenantId("test-tenant")` is not used, unlike other commands.

### Steps to reproduce

See description

### Current behavior

See description

### Expected behavior

Like other commands such a `client.createProcessInstance()`, the configured default tenant id should be used.

### Workaround

Set the tenant id explicitly for this call via `client.newCorrelateMessageCommand().tenantId()`.
      

### Environment

SaaS & SM

### Version

8.8.0-SNAPSHOT

### Rootcause

`CorrelateMessageCommandImpl` does not receive client configuration to copy the configured tenant id.

### Solution ideas

See rootcause. Ideally, we'd find a way to enforce this for all commands so that we do not forget.

### Dev -> QA handover

_No response_

### Links

_No response_

### Automated Test Impact

_No response_

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This fixes the behavior for several commands that previously didn't respect the configured default tenant, always requiring users to explicitly set the tenant instead.

This is now enforced on a best-effort manner with an arch unit test.

Also fixes one edge case for the command to create a process instance with result, best explained in the commit message eebb977a5665bc011943863cb2b5e89a5803bb2c.

closes #38157

### CODE REVIEW

File: clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          .areDeclaredInClassesThat()
+          .areAssignableTo(CommandWithTenantStep.class)
+          .and()
+          // Some constructors are only there for backwards compatibility and we can't change them
+          // use the default tenant.
+          .areNotAnnotatedWith(Deprecated.class)
+          .and()
+          // Special case: This command inherits the default tenant id from the
+          // CreateProcessInstanceCommandImpl that it wraps.
+          .areNotDeclaredIn(CreateProcessInstanceWithResultCommandImpl.class)
```
Comment (nicpuppa) [inline clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java:60]: 🔧 I would extract this into a separate constant. e.g `SPECIAL_CASE_COMMAND` and put the comment there if needed (I think the name of the constant is self explecative)

File: clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                  constructor,
+                  String.format(
+                      "Command %s retrieves the default tenant from %s",
+                      commandClass.getName(), CamundaClientConfiguration.class.getName())));
+        }
+      }
+    };
+  }
+
+  private static boolean isGetDefaultTenantIdCall(final JavaMethodCall call) {
```
Comment (nicpuppa) [inline clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java:93]: 🔧 Right now, if a new method in CamundaClientConfiguration is introduced for tenant defaults, you’ll need to update isGetDefaultTenantIdCall. Instead, you might consider a more flexible check:

```
private static final Set<String> DEFAULT_TENANT_METHODS = Set.of(
    "getDefaultTenantId",
    "getDefaultJobWorkerTenantIds"
);

private static boolean isGetDefaultTenantIdCall(JavaMethodCall call) {
    return call.getTargetOwner().isEquivalentTo(CamundaClientConfiguration.class)
        && DEFAULT_TENANT_METHODS.contains(call.getName());
}
```

**Review by nicpuppa — APPROVED** [review]
Thanks @lenaschoenburg, in particular for the ArchUnit test. I left some comments that might help improve the test. Please have a look

**Comment by megglos** [comment]
Note: let's also fix this on the zeebe-client-java and backport (maybe worth a follow-up PR to avoid conflicts when backporting)

File: clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java (hunk 1 above)
Comment (lenaschoenburg) [reply clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java:60]: Let's hope we won't have to extend that list :crossed_fingers: In that case, I prefer the inline definition just because you don't have to jump and look up what a "special case command" actually is.

File: clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java (hunk 2 above)
Comment (lenaschoenburg) [reply clients/java/src/test/java/io/camunda/client/DefaultTenantAwarenessTest.java:93]: I also disagree here, for the same reason as above. I'm hoping we won't add another method for the same purpose. If we do, it's easy enough to quickly refactor.

**Comment by lenaschoenburg** [comment]
I've added similar changes to the `zeebe-client-java`
